python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -454,20 +454,29 @@ def _get_schema(self):
         if 'schema' in self.type:
             return self.type['schema']
         else:
-            if self._get_class_name() == 'pandas.core.frame.DataFrame':
-                import pandas as pd
-                columns, dtypes = self.rdd.map(lambda x: (x.columns, x.dtypes)).first()
-                self.type['schema'] = {'columns': columns, 'dtypes': dtypes}
-                return self.type['schema']
-            return None
+            class_name, schema = self._get_schema_class_name()
```

Comment:
only call `_get_schema_class_name` if `class_name` does not exist, or is pandas dataframe